dotnet/src/Connectors/Connectors.AI.OpenAI/ChatCompletionWithData/AzureOpenAIChatCompletionWithData.cs

```diff
@@ -166,20 +110,46 @@ private static void ValidateMaxTokens(int? maxTokens)
         }
     }
 
-    private async Task<IReadOnlyList<IChatResult>> ExecuteCompletionRequestAsync(
+    private async Task<IReadOnlyList<ChatMessageContent>> InternalGetChatMessageContentsAsync(
         ChatHistory chat,
+        PromptExecutionSettings? executionSettings,
         Kernel? kernel,
-        OpenAIPromptExecutionSettings executionSettings,
         CancellationToken cancellationToken = default)
     {
-        using var request = this.GetRequest(chat, executionSettings, isStreamEnabled: false);
+        var openAIExecutionSettings = OpenAIPromptExecutionSettings.FromExecutionSettingsWithData(executionSettings, OpenAIPromptExecutionSettings.DefaultTextMaxTokens);
+
+        using var request = this.GetRequest(chat, openAIExecutionSettings, isStreamEnabled: false);
         using var response = await this.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
 
         var body = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
 
         var chatWithDataResponse = this.DeserializeResponse<ChatWithDataResponse>(body);
+        var metadata = GetResponseMetadata(chatWithDataResponse);
+
+        return chatWithDataResponse.Choices.Select(choice => new AzureOpenAIWithDataChatMessageContent(choice, metadata)).ToList();
+    }
+
+    private static Dictionary<string, object?> GetResponseMetadata(ChatWithDataResponse chatResponse)
+    {
+        return new Dictionary<string, object?>(5)
+        {
+            { nameof(chatResponse.Id), chatResponse.Id },
+            { nameof(chatResponse.Model), chatResponse.Model },
```

Comment:
Is it possible to make the model Id a first-class citizen in all `ContentBase`-derived classes, since model id is a common property?

Reply:
I think is a fair point, what you think @matthewbolanos @markwallace-microsoft?

So basically if the underlying model don't provide that in the response, at least the connector will.